src/main/scala/com/github/unchama/seichiassist/subsystems/gacha/subsystems/tradesystems/application/actions/Trade.scala

```diff
@@ -0,0 +1,18 @@
+package com.github.unchama.seichiassist.subsystems.gacha.subsystems.tradesystems.application.actions
+
+import com.github.unchama.seichiassist.subsystems.gacha.subsystems.tradesystems.domain.TradeResult
+
+trait Trade[F[_], ItemStack] {
+
+  /**
+   * ガチャアイテムから椎名林檎やガチャ券へ交換できるアイテムを列挙する
+   */
+  def trade(contents: List[ItemStack]): F[TradeResult[ItemStack]]
+
+}
```

Comment:
これはドメインオブジェクトであり(つまり `domain` パッケージに入れてよい)、

```suggestion
/**
 * 特定のアイテムを交換するためのルール（何を何個入れると何が何個得られて何が返却されるか）
 * を定めるオブジェクトの trait
 */
trait TradeRule[ItemStack] {

  /**
   * ガチャアイテムから椎名林檎やガチャ券へ交換できるアイテムを列挙する
   */
  def outcomeWhenGiven(items: List[ItemStack]): TradeResult[ItemStack]

}
```

のような `trait` にするのが適切かなと思います。

2022/08/10 深夜に rito さんと話したこと：
 - `TradeRule` は純粋関数を規定するので、 `F[_]` は不要
 - 「有象無象のアイテムスタックを受け取って何らかのアイテムスタックを受け取る」という、「ガチャのアイテムを受け取って特定のアイテムスタックを返す」より抽象的な見方をすると、これは `gacha` subsystem の外にあるべき別の `subsystem` であるという結論に至った